src/erf.jl

```diff
@@ -0,0 +1,265 @@
+pa4  = -1.10894694282396677476e-01 # 0xBFBC6398, 0x3D3E28EC
+pa5  =  3.54783043256182359371e-02 # 0x3FA22A36, 0x599795EB
+pa6  = -2.16637559486879084300e-03 # 0xBF61BF38, 0x0A96073F
+qa1  =  1.06420880400844228286e-01 # 0x3FBB3E66, 0x18EEE323
+qa2  =  5.40397917702171048937e-01 # 0x3FE14AF0, 0x92EB6F33
+qa3  =  7.18286544141962662868e-02 # 0x3FB2635C, 0xD99FE9A7
+qa4  =  1.26171219808761642112e-01 # 0x3FC02660, 0xE763351F
+qa5  =  1.36370839120290507362e-02 # 0x3F8BEDC2, 0x6B51DD1C
+qa6  =  1.19844998467991074170e-02 # 0x3F888B54, 0x5735151D
+
+# Coefficients for approximation to  erfc in [1.25,1/0.35]
+ra0  = -9.86494403484714822705e-03 # 0xBF843412, 0x600D6435 
+ra1  = -6.93858572707181764372e-01 # 0xBFE63416, 0xE4BA7360 
+ra2  = -1.05586262253232909814e+01 # 0xC0251E04, 0x41B0E726 
+ra3  = -6.23753324503260060396e+01 # 0xC04F300A, 0xE4CBA38D 
+ra4  = -1.62396669462573470355e+02 # 0xC0644CB1, 0x84282266 
+ra5  = -1.84605092906711035994e+02 # 0xC067135C, 0xEBCCABB2 
+ra6  = -8.12874355063065934246e+01 # 0xC0545265, 0x57E4D2F2 
+ra7  = -9.81432934416914548592e+00 # 0xC023A0EF, 0xC69AC25C 
+sa1  =  1.96512716674392571292e+01 # 0x4033A6B9, 0xBD707687 
+sa2  =  1.37657754143519042600e+02 # 0x4061350C, 0x526AE721 
+sa3  =  4.34565877475229228821e+02 # 0x407B290D, 0xD58A1A71 
+sa4  =  6.45387271733267880336e+02 # 0x40842B19, 0x21EC2868 
+sa5  =  4.29008140027567833386e+02 # 0x407AD021, 0x57700314 
+sa6  =  1.08635005541779435134e+02 # 0x405B28A3, 0xEE48AE2C 
+sa7  =  6.57024977031928170135e+00 # 0x401A47EF, 0x8E484A93 
+sa8  = -6.04244152148580987438e-02 # 0xBFAEEFF2, 0xEE749A62 
+
+# Coefficients for approximation to  erfc in [1/.35,28]
+rb0  = -9.86494292470009928597e-03 # 0xBF843412, 0x39E86F4A
+rb1  = -7.99283237680523006574e-01 # 0xBFE993BA, 0x70C285DE
+rb2  = -1.77579549177547519889e+01 # 0xC031C209, 0x555F995A
+rb3  = -1.60636384855821916062e+02 # 0xC064145D, 0x43C5ED98
+rb4  = -6.37566443368389627722e+02 # 0xC083EC88, 0x1375F228
+rb5  = -1.02509513161107724954e+03 # 0xC0900461, 0x6A2E5992
+rb6  = -4.83519191608651397019e+02 # 0xC07E384E, 0x9BDC383F
+sb1  =  3.03380607434824582924e+01 # 0x403E568B, 0x261D5190
+sb2  =  3.25792512996573918826e+02 # 0x40745CAE, 0x221B9F0A
+sb3  =  1.53672958608443695994e+03 # 0x409802EB, 0x189D5118
+sb4  =  3.19985821950859553908e+03 # 0x40A8FFB7, 0x688C246A
+sb5  =  2.55305040643316442583e+03 # 0x40A3F219, 0xCEDF3BE6
+sb6  =  4.74528541206955367215e+02 # 0x407DA874, 0xE79FE763
+sb7  = -2.24409524465858183362e+01 # 0xC03670E2, 0x42712D62
+
+function erfc1(x::Float64)
+	s = fabs(x) - 1 
+	P = pa0+s*(pa1+s*(pa2+s*(pa3+s*(pa4+s*(pa5+s*pa6)))))
+	Q = 1+s*(qa1+s*(qa2+s*(qa3+s*(qa4+s*(qa5+s*qa6)))))
+	return 1 - erx - P/Q
+end
+
+function erfc2(ix::UInt32, x::Float64)
+
+	if ix < 0x3ff40000  # |x| < 1.25
+		return erfc1(x)
+	end
+
+	x = fabs(x) 
+	s = 1/(x*x)
+	if ix < 0x4006db6d # |x| < 1/.35 ~ 2.85714
+		R = ra0+s*(ra1+s*(ra2+s*(ra3+s*(ra4+s*(ra5+s*(ra6+s*ra7))))))
```

Comment:
You can use the `Base.Math.@horner` macro for these things: it has the nice side-effect that it uses `muladd`, which can make use of FMA instructions on architectures which have it (e.g. Intel Haswell and newer).